pkgs/applications/editors/vim/plugins/blink-cmp/default.nix

```diff
@@ -32,6 +35,16 @@ vimUtils.buildVimPlugin {
     mkdir -p target/release
     ln -s ${blink-fuzzy-lib}/lib/libblink_cmp_fuzzy.${libExt} target/release/libblink_cmp_fuzzy.${libExt}
   '';
+
+  passthru = {
+    updateScript = nix-update-script {
+      attrPath = "vimPlugins.blink-cmp.blink-fuzzy-lib";
+    };
```

Comment:
I believe it's unnecessary to specify the `attrPath` as I was able to successfully update without it:

```suggestion
    updateScript = nix-update-script { };
```

Reply:
No... 
<details>

<summary>Git diff without attr path</summary>

`nix-update vimPlugins.blink-cmp`

```diff
diff --git a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
index 3da1bfde9822..4d7f438c31fa 100644
--- a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
+++ b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
@@ -7,12 +7,12 @@
   nix-update-script,
 }:
 let
-  version = "0.6.2";
+  version = "0.7.1";
   src = fetchFromGitHub {
     owner = "Saghen";
     repo = "blink.cmp";
     rev = "refs/tags/v${version}";
-    hash = "sha256-uAIO8Q9jQvZ36Ngz9dURH7Jo7+WX/swauhj3ltIkZ5c=";
+    hash = "sha256-IHl+XIldo2kculpbiOuLIJ6RJbFODiRlQU4x8hvE7pI=";
   };
   libExt = if stdenv.hostPlatform.isDarwin then "dylib" else "so";
   blink-fuzzy-lib = rustPlatform.buildRustPackage {
```
</details>

<details>
<summary>Git diff with attr path</summary>

`nix-update vimPlugins.blink-cmp.blink-fuzzy-lib` ([src](https://github.com/NixOS/nixpkgs/blob/e3ca8012e9514272e5c7959768193a68e0bf1521/pkgs/by-name/ni/nix-update/nix-update-script.nix#L11))

```diff
diff --git a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
index 3da1bfde9822..3d061ddb45cc 100644
--- a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
+++ b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
@@ -7,12 +7,12 @@
   nix-update-script,
 }:
 let
-  version = "0.6.2";
+  version = "0.7.1";
   src = fetchFromGitHub {
     owner = "Saghen";
     repo = "blink.cmp";
     rev = "refs/tags/v${version}";
-    hash = "sha256-uAIO8Q9jQvZ36Ngz9dURH7Jo7+WX/swauhj3ltIkZ5c=";
+    hash = "sha256-IHl+XIldo2kculpbiOuLIJ6RJbFODiRlQU4x8hvE7pI=";
   };
   libExt = if stdenv.hostPlatform.isDarwin then "dylib" else "so";
   blink-fuzzy-lib = rustPlatform.buildRustPackage {
@@ -20,7 +20,7 @@ let
     pname = "blink-fuzzy-lib";
 
     useFetchCargoVendor = true;
-    cargoHash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
+    cargoHash = "sha256-XXI2jEoD6XbFNk3O8B6+aLzl1ZcJq1VinQXb+AOw8Rw=";
 
     env = {
       # TODO: remove this if plugin stops using nightly rust
```

</details>

First one ignores `cargoHash` which may cause a build failure if it changes (I reset it to show the difference)

Reply:
I'm using `nix-update` from the latest nixpkgs and it's updating `cargoHash` without specifying the `attrPath`:

```console
nix run github:NixOS/nixpkgs#nix-update vimPlugins.blink-cmp.blink-fuzzy-lib
```

<details><summary>Git diff without attr path</summary>
<p>

```diff
---
 pkgs/applications/editors/vim/plugins/blink-cmp/default.nix | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
index b4a3f3cf1125..60a8ee55ca57 100644
--- a/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
+++ b/pkgs/applications/editors/vim/plugins/blink-cmp/default.nix
@@ -7,12 +7,12 @@
   nix-update-script,
 }:
 let
-  version = "0.6.2";
+  version = "0.7.1";
   src = fetchFromGitHub {
     owner = "Saghen";
     repo = "blink.cmp";
     rev = "refs/tags/v${version}";
-    hash = "sha256-uAIO8Q9jQvZ36Ngz9dURH7Jo7+WX/swauhj3ltIkZ5c=";
+    hash = "sha256-IHl+XIldo2kculpbiOuLIJ6RJbFODiRlQU4x8hvE7pI=";
   };
   libExt = if stdenv.hostPlatform.isDarwin then "dylib" else "so";
   blink-fuzzy-lib = rustPlatform.buildRustPackage {
@@ -20,7 +20,7 @@ let
     pname = "blink-fuzzy-lib";
 
     useFetchCargoVendor = true;
-    cargoHash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
+    cargoHash = "sha256-XXI2jEoD6XbFNk3O8B6+aLzl1ZcJq1VinQXb+AOw8Rw=";
 
     env = {
       # TODO: remove this if plugin stops using nightly rust
-- 
```

</p>
</details>

Reply:
`nix-update` doesn't use `passthru.updateScipt`, it is used by [r-ryantm](https://nix-community.github.io/nixpkgs-update/r-ryantm/#r-ryantm) to figure out what arguments pass to `nix-update`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#automatic-package-updates

```
~/dev/nixpkgs/review master ❯ gh pr checkout 358078
~/dev/nixpkgs/review add-blink-update-script ❯ nix repl
Lix 2.92.0-dev-pre20241129-2e5780e
Type :? for help.
nix-repl> :l .
Added 23011 variables.

nix-repl> vimPlugins.blink-cmp.passthru.updateScript
[
  "/nix/store/mj2yymgya1s5hqvpm7k1y4vxlzy26qxw-nix-update-1.7.0/bin/nix-update"
  "vimPlugins.blink-cmp.blink-fuzzy-lib"
]
```

Reply:
You're right. I usually use `nix-shell maintainers/scripts/update.nix`, but I though `nix-update` does the same by default, which turns out you also need to specify the `--use-update-script` flag for that. Thanks a lot for the info!